src/wallet/spend.cpp

```diff
@@ -1557,6 +1557,69 @@ std::pair<unsigned int, bool> CalculateDeniabilizationCycles(CWallet& wallet, co
     return std::make_pair(inputStats.first + deniabilizationCycles, inputStats.second);
 };
 
+void SpoofTransactionFingerprint(CMutableTransaction& tx, FastRandomContext& rng_fast)
+{
+    // Transaction "fingerprint" spoofing
+    struct Fingerprint {
+        bool standardVersion = false;
+        bool antiFeeSniping = false;
+        bool bip69Ordering = false;
+    };
+
+    // wallet fingerprints based on info from variuous sources, see:
+    // https://github.com/achow101/wallet-fingerprinting/blob/main/fingerprints.md
+    // https://gitlab.com/1440000bytes/goldfish
+    // https://ishaana.com/blog/wallet_fingerprinting/
+    static const Fingerprint s_walletFingerprints[] = {
+        {true, true, false},   // Core
+        {true, true, true},    // Electrum
+        {true, false, false},  // Blue
+        {false, false, true},  // Trezor
+        {false, false, false}, // Trust, Ledger
+    };
```

Comment:
Can also [randomize RBF](https://github.com/btcpayserver/btcpayserver/pull/1596) to make fingerprinting more difficult.

Reply:
I was thinking about randomizing RBF - however we issue the deniabilization transactions with the lowest fee possible (eg confirm_target = 100+ for daily frequency, 1000+ for weekly frequency), so there's a chance the TX may get stuck if the mempool suddenly gets crowded. If that happens we attempt an RBF at said frequency, so I think it's important to have it always enabled.
Alternatively we could randomize RBF, and use a conservative fee calculation in the RBF=off case, but that would likely be more expensive ...

Reply:
- Transactions can be replaced even if they do not signal RBF since some nodes and miners support full RBF
- I don't think using core fee estimation for these transactions is a good idea

Reply:
Ah, interesting, yeah if we can rely on full-RBF nodes, then we can randomize the flag.
Is Core's fee estimation also fingerprint-able? I could randomize that too, but what is a good alternative estimation?

Reply:
I added RBF spoofing in the latest commit.

Reply:
Hmm, so it doesn't look like feebumper::CommitTransaction would let me bump a transaction if the RBF flag wasn't set on the original transaction. For now I worked around it by passing an extra flag into SpoofTransactionFingerprint to only signal no-rbf when RBF is not required (as is the case with deniabilizaiton transactions). Let me know if you have better ideas how to work around this and take advantage of full-RBF nodes bumping regardless of the flag ...